kernel/src/engine/default/json.rs

```diff
@@ -133,14 +133,14 @@ impl<E: TaskExecutor> JsonHandler for DefaultJsonHandler<E> {
         // This channel will become the output iterator
         // The stream will execute in the background, and we allow up to `readahead`
         // batches to be buffered in the channel.
-        let (sender, receiver) = std::sync::mpsc::sync_channel(self.readahead);
+        let (sender, mut receiver) = tokio::sync::mpsc::channel(self.readahead);
```

Comment:
If we stick with an async channel, we should use [`async_channel`](https://docs.rs/async-channel/latest/async_channel/) rather than the tokio one, again so we don't depend on a particular runtime. (perhaps the tokio one doesn't depend on the rest of tokio and is okay as well, we'd need to make sure)